core/src/main/java/kafka/server/share/SharePartitionManager.java

```diff
@@ -706,6 +712,16 @@ private SharePartitionKey sharePartitionKey(String groupId, TopicIdPartition top
         return new SharePartitionKey(groupId, topicIdPartition);
     }
 
+    /**
+     *
+     * @param groupId The share group id, this is used to identify the share group.
+     * @param topicIdPartition The topic partition that the group is subscribed to.
+     * @return The share partition stored for the share group topic-partition.
+     */
+    protected SharePartition fetchSharePartition(String groupId, TopicIdPartition topicIdPartition) {
```

Comment:
Unlike SharePartition fetchSharePartition(SharePartitionKey sharePartitionKey), this method can return null. Should all callers check for null explicitly?

Could we name them differently so that the semantic is clear? Perhaps `getOrCreateSharePartition` and `sharePartition`?